src/main/steps/common/keep-details-private/KeepYourDetailsPrivateMapper.ts

```diff
@@ -1,65 +1,76 @@
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 import { CaseWithId } from '../../../app/case/case';
-import { ConfidentialityList, KeepDetailsPrivate, PartyDetails, YesOrNo } from '../../../app/case/definition';
+import { ConfidentialityList, PartyDetails, YesOrNo } from '../../../app/case/definition';
 import type { AppRequest } from '../../../app/controller/AppRequest';
 
-export const setKeepYourDetailsPrivate = (partyDetails: PartyDetails, req: AppRequest): PartyDetails => {
-  let keepDetailsPrivate: KeepDetailsPrivate;
+export const prepareKeepDetailsPrivateRequest = (req: AppRequest): Partial<PartyDetails> => {
+  const { startAlternative, contactDetailsPrivate, detailsKnown } = req.session.userCase;
+  const request: Partial<PartyDetails> = {};
+
   const confidentialityList: ConfidentialityList[] = [];
 
-  if (req.session.userCase.startAlternative === YesOrNo.YES && req?.session?.userCase?.contactDetailsPrivate) {
-    req.session.userCase.contactDetailsPrivate.forEach(element => {
+  if (startAlternative === YesOrNo.YES && contactDetailsPrivate) {
+    contactDetailsPrivate.forEach(element => {
       confidentialityList.push(ConfidentialityList[element]);
     });
   }
 
-  if (partyDetails.response && partyDetails.response?.keepDetailsPrivate) {
-    keepDetailsPrivate = partyDetails.response?.keepDetailsPrivate;
-    keepDetailsPrivate.otherPeopleKnowYourContactDetails = req.session.userCase.detailsKnown!;
-    keepDetailsPrivate.confidentiality = req.session.userCase.startAlternative!;
-    keepDetailsPrivate.confidentialityList = confidentialityList;
-  } else {
-    partyDetails.response = {
+  Object.assign(request, {
+    response: {
```

Comment:
Why do we need to group all the data under response.? Assign those keys directly to request and then use the request and assign it to response appropriately in the controllers.

Reply:
Done, the keys are now being assigned to the request directly